homeassistant/components/mqtt/models.py

```diff
@@ -231,11 +231,23 @@ def async_render_with_possible_json_value(
                 values,
                 self._value_template,
             )
-            rendered_payload = (
-                self._value_template.async_render_with_possible_json_value(
-                    payload, variables=values
+            try:
+                rendered_payload = (
+                    self._value_template.async_render_with_possible_json_value(
+                        payload, variables=values
+                    )
                 )
-            )
+            except Exception as ex:  # pylint: disable=broad-except
```

Comment:
I think we can be more specific and use `TemplateError`?

Reply:
I meant you can replace the broad `Exception` to `TemplateError` as I think the function within can't raise anything else?

Reply:
That is not correct. As you can see in the test it can raise a TypeError

Reply:
Right. Wasn't sure if that was caught and returned as `TemplateError` or not.
But then we should probably re-raise same exception instead of changing.

Reply:
I merge out the last commit